File: frontend/src/types/mcp.ts
// MCP 模块类型定义

// MCP 环境变量
export interface McpEnv {
  k: string  // key
  v: string  // value
}

// MCP 服务
export interface Mcp {
  id: string
  name: string
  tp: 'stdio' | 'SSE'  // 类型：stdio 或 SSE
  cmd: string  // 命令或 URL
  status: 'on' | 'off' | 'err'  // 状态：运行中/停止/错误
  toolCount: number  // 工具数量
  env: McpEnv[]  // 环境变量列表
  timeout: number  // 超时时间（秒）
  createdAt?: string
}

// MCP 测试结果
export interface McpTestResult {
  success: boolean
  toolCount: number
  tools?: string[]  // 工具名称列表
  error?: string
  duration: number
}

File: backend/app/schemas/mcp.py
"""MCP 模块 Pydantic schema（对齐前端 types/mcp.ts）。"""
from pydantic import BaseModel


class McpEnv(BaseModel):
    k: str
    v: str


class McpCreate(BaseModel):
    name: str
    tp: str = "stdio"
    cmd: str = ""
    status: str = "off"
    toolCount: int = 0
    env: list[McpEnv] = []
    timeout: int = 30


class McpUpdate(BaseModel):
    name: str | None = None
    tp: str | None = None
    cmd: str | None = None
    status: str | None = None
    toolCount: int | None = None
    env: list[McpEnv] | None = None
    timeout: int | None = None


class McpOut(BaseModel):
    id: str
    name: str
    tp: str
    cmd: str
    status: str
    toolCount: int
    env: list[McpEnv]
    timeout: int
    createdAt: str | None = None


class McpTestResult(BaseModel):
    success: bool
    toolCount: int
    tools: list[str] | None = None
    error: str | None = None
    duration: float
